### ISSUE TITLE
Errors after upgrading to latest vcpkg version

### ISSUE THREAD

**Issue by gamagan** [gabime/spdlog#2878]
Hello,
I've been using an older version of spdlog. Recently, I wanted to update that dependency via `vcpkg`, so I installed the latest vcpkg version, which 1.12.0. 
After installing that, I now started to get compilation errors due to the call to the macro below.

```
#define my_LOG_TRACE(fmt, ...) {\
    spdlog::get("logger-trace")->info(std::string(std::string(__FUNCTION__) + "()> " + std::string(fmt)).data(), ## __VA_ARGS__); \
}
```

Usage of that log would be something like:
```
my_LOG_TRACE("Recreating Swapchain -- count:{}, w:{}, h:{}.", mSwapchainCount, mWindowW, mWindowH);
```

The errors are of the form:
```
error C7595: 'fmt::v10::basic_format_string<char,ngen::u8 &,ngen::u32 &,ngen::u32 &>::basic_format_string': call to immediate function is not a constant expression
```

I'm not sure how to fix this error other than reverting.

**Comment by tt4g** [comment]
Duplicate fmtlib/fmt#2931 and #2430

Workaround: use `fmt::runtime()`.

**Comment by gamagan** [comment]
> Duplicate [fmtlib/fmt#2931](https://github.com/fmtlib/fmt/issues/2931) and #2430
> 
> Workaround: use `fmt::runtime()`.

I appreciate the suggestion. 
I tried adding that `fmt::runtime()`, but it creates a whole new set of errors. It also requires me to include the fmt header for that function everywhere I have logging.

I just ended up reverting the upgrade and will stay on the version I'm on now. I'll try again in the future to see if these issues disappear and I could upgrade in peace. This is a cool library, but I've never been able to upgrade it by just dropping in the new code. Something always breaks in the process.
